realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -264,6 +290,78 @@ public boolean hasField(String fieldName) {
     }
 
     /**
+     * Set the value for the given field. This method will automatically try to convert numbers and
+     * booleans that are given as {@code String} to their appropriate type. E.g. {@code "10"} will be
+     * converted to {@code 10} if the field type is {@code int}.
+     *
+     * Using the typed setters will be faster than using this method.
+     *
+     * @throws IllegalArgumentException if field name doesn't exists or if the input value cannot be converted
+     * to the appropriate input type.
+     * @throws NumberFormatException if a String based number cannot be converted properly.
+     */
+    @SuppressWarnings("unchecked")
+    public void set(String fieldName, Object value) {
+        boolean isString = (value instanceof String);
+        String strValue = isString ? (String) value : null;
+
+        // Do implicit conversion if needed
+        long columnIndex = row.getColumnIndex(fieldName);
+        RealmFieldType type = row.getColumnType(columnIndex);
+        if (isString && type != RealmFieldType.STRING) {
+            switch(type) {
+                case BOOLEAN: value = Boolean.parseBoolean(strValue); break;
+                case INTEGER: value = Long.parseLong(strValue); break;
+                case FLOAT: value = Float.parseFloat(strValue); break;
+                case DOUBLE: value = Double.parseDouble(strValue); break;
+                default:
+                    throw new IllegalArgumentException(String.format("Field %s is not a String field, " +
+                            "and the provide value could not be automatically converted: %s. Use a typed" +
+                            "setter instead", fieldName, value));
+            }
+        }
+
+        if (value == null) {
+            row.nullifyLink(row.getColumnIndex(fieldName));
+            // TODO Add support for other types when Null is merged.
+        } else if (value instanceof Boolean) {
```

Comment:
I would go further than that since lots of equals can be costly as well.

We could setup an `Enum` and mentioned above. Let's call it SupportedType.
Then we could have a `static Map<Class<?>, SupportedType>`
Then we could do:

```java
SupportedType type = Types.get(value.getClass());
if (type != null) {
   type.setField(fieldName, value);
} else {
   // blah blah...
}
```